public/js/index.js

```diff
@@ -0,0 +1,61 @@
+async function retrieveJSONLength() {
+  getProduct(await productService.getProductLength())
+}
+
+async function getProduct(num, amount = 5) {
+  //amount is how many products will load.
+  let productIds = []
```

Comment:
I think we can get rid of `retreiveJSONLength()`.

```suggestion
  let productIds = []
  let totalProducts = (await this.productService.getProducts()).length;
```